### DISCUSSION TITLE
Use expression_v2 with arctan2

### DISCUSSION THREAD

**Discussion by IvanBioli** [evalf/nutils#888] [Q&A]
Hi,

is there any way to use the expression parser version 2, i.e. `expression_v2` with the function `arctan2`, which takes two arguments? I read in the documentation of `expression_v2`that "A function takes a **single** argument", so I was wandering if there is any workaround.

Best,
Ivan

**Comment by joostvanzwieten** [comment]
Hi Ivan,

The problem with functions of multiple arguments is that we don't know what to do with the indices. Should the behaviour be the same as for an addition or multiplication (or something else)? The only unambiguous situation is where only one of the arguments has indices and the other none, e.g. `arctan2(u_ij, v)`, but we haven't implemented that.

The expression syntax is not, and will never be, complete. If something cannot be written using the expression syntax, you have to use NumPy array style operations.

Best, Joost
